src/gt4py/next/otf/compilation/build_systems/compiledb.py

```diff
@@ -227,29 +227,42 @@ def _cc_prototype_program_source(
     )
 
 
-def _cc_find_compiledb(
-    prototype_program_source: stages.ProgramSource, cache_lifetime: config.BuildCacheLifetime
-) -> Optional[pathlib.Path]:
+def _cc_get_compiledb(
+    renew_compiledb: bool,
+    prototype_program_source: stages.ProgramSource,
+    build_type: config.CMakeBuildType,
+    cmake_flags: list[str],
+    cache_lifetime: config.BuildCacheLifetime,
+) -> pathlib.Path:
     cache_path = cache.get_cache_folder(
         stages.CompilableSource(prototype_program_source, None), cache_lifetime
     )
-    compile_db_path = cache_path / "compile_commands.json"
+
+    with lock.Lock(str(cache_path / "compiledb.lock"), lifetime=120):  # type: ignore[attr-defined] # mypy not smart enough to understand custom export logic
```

Comment:
It may be worth to add a  comment explaining why a lock is needed here and the chosen lifetime value